minetest/minetest
# [No Squash] Show dep errors in Select Mods modal

- Refactor ModConfiguration to be a reusable component rather than being inherited
- Use ModConfiguration in Select Mods dialog to show errors

Fixes #9878
Fixes #11712

Related #3330

Missing dependency:

![image](https://user-images.githubusercontent.com/2122943/167272507-33e5d8a5-e32f-4756-a7ed-88e232bd22ae.png)

Cycle:

![image](https://user-images.githubusercontent.com/2122943/167272515-2a6c9912-658e-4ede-a658-694ff32b0b1a.png)

Errors in mod packs and games

![image](https://user-images.githubusercontent.com/2122943/167278461-2a79f660-2f70-43d6-8791-3965e281ade5.png)

## To do

- [ ] Add/modify unit tests
- [x] Show errors on modpacks/games

## How to test

- Check the Select Mod behaviour when:
    - A dep isn't installed
    - A dep isn't enabled
    - There's a cycle
- Check that "The following mods could not be found:" still works

https://github.com/minetest/minetest/files/8646053/mods.zip
